src/adapter/src/catalog/storage.rs

```diff
@@ -112,12 +114,18 @@ async fn migrate<S: Append>(
             )?;
             txn.id_allocator.insert(
                 IdAllocKey {
-                    name: ROLE_ID_ALLOC_KEY.into(),
+                    name: USER_ROLE_ID_ALLOC_KEY.into(),
```

Comment:
If this is a breaking change to which we aren't providing an upgrade path, the cloud team should be flagged.

Reply:
https://materializeinc.slack.com/archives/CL68GT3AT/p1660169050996079